Assets/Scripts/UnityServices/Lobbies/LobbyServiceFacade.cs

```diff
@@ -264,6 +225,68 @@ public async Task<(bool Success, Lobby Lobby)> TryQuickJoinLobbyAsync()
             return (false, null);
         }
 
+        void ResetLobby()
+        {
+            CurrentUnityLobby = null;
+            m_LocalUser.ResetState();
+            m_LocalLobby?.Reset(m_LocalUser);
+
+            // no need to disconnect Netcode, it should already be handled by Netcode's callback to disconnect
+        }
+
+        void OnLobbyChanges(ILobbyChanges changes)
+        {
+            if (changes.LobbyDeleted)
+            {
+                Debug.Log("Lobby deleted");
+                ResetLobby();
+            }
+            else
+            {
+                Debug.Log("Lobby updated");
+                changes.ApplyToLobby(CurrentUnityLobby);
+                m_LocalLobby.ApplyRemoteData(CurrentUnityLobby);
+
+                // as client, check if host is still in lobby
+                if (!m_LocalUser.IsHost)
+                {
+                    foreach (var lobbyUser in m_LocalLobby.LobbyUsers)
+                    {
+                        if (lobbyUser.Value.IsHost)
+                        {
+                            return;
+                        }
+                    }
+
+                    m_UnityServiceErrorMessagePub.Publish(new UnityServiceErrorMessage("Host left the lobby", "Disconnecting.", UnityServiceErrorMessage.Service.Lobby));
+                    EndTracking();
+                    // no need to disconnect Netcode, it should already be handled by Netcode's callback to disconnect
+                }
+            }
+        }
+
+        void OnKickedFromLobby()
+        {
+            Debug.Log("Kicked from Lobby");
+            ResetLobby();
+        }
+
+        void OnLobbyEventConnectionStateChanged(LobbyEventConnectionState lobbyEventConnectionState)
+        {
+            m_LobbyEventConnectionState = lobbyEventConnectionState;
+            Debug.Log($"LobbyEventConnectionState changed to {lobbyEventConnectionState}");
+        }
+
+        async void SubscribeToJoinedLobby()
+        {
+            var lobbyEventCallbacks = new LobbyEventCallbacks();
+            lobbyEventCallbacks.LobbyChanged += OnLobbyChanges;

```

Comment:
mmm I'm scared of those callbacks with no associated -=
I'm assuming the below SubscribeToLobby takes ownership and just handles that for you?
(You can resolve this shouldn't be a blocking comment)

Reply:
Yes. The `LobbyEventCallbacks` object is passed to the `LobbyChannel` that is created internally by the Lobby SDK. This is then stored in `m_LobbyEvents`. The callbacks we register here are called by other callbacks managed internally by the `LobbyChannel` object. These are unsubscribed to when we call `m_LobbyEvents.UnsubscribeAsync()` and the `LobbyEventCallbacks` object gets garbage collected. When we subscribe again, we pass in a new `LobbyEventCallbacks` object.